microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/SubsystemAdd.java

```diff
@@ -19,41 +19,60 @@
  * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
  * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
  */
-
 package org.wildfly.extension.microprofile.opentracing;
 
+import static org.wildfly.extension.microprofile.opentracing.SubsystemDefinition.DEFAULT_TRACER;
+import static org.wildfly.extension.microprofile.opentracing.SubsystemDefinition.DEFAULT_TRACER_CAPABILITY;
+
 import org.jboss.as.controller.AbstractBoottimeAddStepHandler;
 import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.registry.Resource;
 import org.jboss.as.server.AbstractDeploymentChainStep;
 import org.jboss.as.server.DeploymentProcessorTarget;
 import org.jboss.as.server.deployment.Phase;
 import org.jboss.dmr.ModelNode;
+import org.wildfly.extension.microprofile.opentracing.resolver.JaegerEnvTracerConfiguration;
+import org.wildfly.microprofile.opentracing.smallrye.WildFlyTracerFactory;
 
 class SubsystemAdd extends AbstractBoottimeAddStepHandler {
+
     static final SubsystemAdd INSTANCE = new SubsystemAdd();
 
     private SubsystemAdd() {
-        super();
+        super(DEFAULT_TRACER);
     }
 
     @Override
-    protected void performBoottime(OperationContext context, ModelNode operation, ModelNode model) {
+    protected void performBoottime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+        WildFlyTracerFactory.registerCompatibilityTracer(new JaegerEnvTracerConfiguration());
         TracingExtensionLogger.ROOT_LOGGER.activatingSubsystem();
         context.addStep(new AbstractDeploymentChainStep() {
+            @Override
             public void execute(DeploymentProcessorTarget processorTarget) {
                 processorTarget.addDeploymentProcessor(
                         SubsystemExtension.SUBSYSTEM_NAME,
                         Phase.DEPENDENCIES,
                         Phase.DEPENDENCIES_MICROPROFILE_OPENTRACING,
                         new TracingDependencyProcessor()
                 );
-
                 processorTarget.addDeploymentProcessor(SubsystemExtension.SUBSYSTEM_NAME,
                         Phase.POST_MODULE,
                         Phase.POST_MODULE_MICROPROFILE_OPENTRACING,
                         new TracingDeploymentProcessor()
                 );
             }
         }, OperationContext.Stage.RUNTIME);
+        super.performBoottime(context, operation, model);
+    }
+
+    @Override
+    protected void recordCapabilitiesAndRequirements(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+        super.recordCapabilitiesAndRequirements(context, operation, resource);
```

Comment:
I don't think you want to call the super implementation, as you are handling capability registration conditionally.